### PR TITLE
Enhance hammer_ping for 1.x

### PR DESCRIPTION
**Pull request by xiangce**
- now can handle some unexpected outputs
- update interfaces

### CODE REVIEW

File: insights/parsers/hammer_ping.py [hunk 1]
```diff
@@ -1,21 +1,13 @@
 """
-hammer_ping - command ``/usr/bin/hammer ping``
-==============================================
+HammerPing - command ``/usr/bin/hammer ping``
+=============================================
 
 The hammer ping parser reads the output of ``hammer ping`` and turns it into
 a dictionary of service statuses in the ``data`` property.  Each service has
```
Comment (bfahr) [inline insights/parsers/hammer_ping.py]: @xiangce the `data` attribute is mentioned here, but not shown in the examples or defined as an attribute.  Would you mind to add docstrings for the two attributes `data` and `error`?

Comment (xiangce) [inline insights/parsers/hammer_ping.py]: @bfahr - Sure, I will modify the docstrings in next commit. Thanks.

File: insights/parsers/hammer_ping.py [hunk 2]
```diff
@@ -1,21 +1,13 @@
...
-its status stored in the 'status' key and its response in the 'repsonse' key;
+its status stored in the 'status' key and its response in the 'respsonse' key;
```
Comment (xiangce) [inline insights/parsers/hammer_ping.py]: Thanks Paul

File: insights/parsers/hammer_ping.py [hunk 3]
```diff
@@ -31,68 +23,77 @@
...
-    def is_ok(self, service):
-        """
-        (bool): is the service defined in the data and is it listed as 'ok'?
-        """
-        return service in self.data and 'ok' == self.data[service]['status']
+    @property
+    def is_ok(self):
+        """bool: Are all services OK?"""
+        if not self.errors:
+            if self.data and all(svc['status'].lower() == 'ok' for svc in self.data.values()):
```
Comment (xiangce) [inline insights/parsers/hammer_ping.py]: Make sense, will do it

File: insights/parsers/hammer_ping.py (hunk 2 above)
Comment (xiangce) [inline insights/parsers/hammer_ping.py]: Your code looks good to me, I will use it directly.

**Comment by xiangce** [comment]
@PaulWay - Would you please to check this PR again? If you think it's OK, please approve it then I can merge it. Thanks.

File: insights/parsers/hammer_ping.py [hunk 4]
```diff
@@ -30,69 +22,75 @@
...
-        self.data = dic
+        self.status_of_service = {}
+        """dict: The status of each service, converted to lower case."""
+        self.response_of_service = {}
+        """dict: The response of each service, as is."""
+        self.service_list = []
+        """list: The list of service names in order."""
+        self.errors = []
+        """list: Any error messages encountered during parsing."""
+        self.is_ok = False
```
Comment (xiangce) [inline insights/parsers/hammer_ping.py]: @bfahr - OK, will do it in next commit.

File: insights/parsers/hammer_ping.py [hunk 5]
```diff
@@ -30,69 +22,75 @@
...
+        """bool: Were all services read correctly and are they all in 'ok' state?"""
```
Comment (xiangce) [inline insights/parsers/hammer_ping.py]: My fault, will do it. Thanks a lot.

**Comment by xiangce** [comment]
@bfahr - I reverted the removed APIs and marked them as deprecated in docstring. Please check it again.

BTW, in order to mark the "self.data" individually, I added the "__init__" function.

**Pull request merged**